deephaven/deephaven-js-plugins
# Add plotly plugin

## Add Plotly JS plugin
Display plotly plots in the Web UI. Requires [deephaven-plugin-plotly](https://pypi.org/project/deephaven-plugin-plotly/) to be installed on the server.

## Testing locally
- Set up local Deephaven server using [deephaven-server as a package](https://pypi.org/project/deephaven-server/)
- Install [deephaven-plugin-plotly](https://pypi.org/project/deephaven-plugin-plotly/): `pip i deephaven-plugin-plotly`
- Start the server on port `9000`
```
from deephaven_server import Server
server = Server(port=9000)
server.start()
```
- Clone deephaven-js-plugins (this repository)
- Install deephaven-js-plugins dependencies and build the plugins: `npm i; npm run build`
- Add `manifest.json` to the root of the repo:
```
{
  "plugins": [
    { "name": "plotly", "main": "dist/index.js" }
  ]
}
```
- Serve static files from the parent folder or this repo:`npx serve ./ -l 9010`
- Open http://localhost:9010 and confirm it serves the folder containing `deephaven-js-plugins` with the manifest
- Clone [web-client-ui repo](https://github.com/deephaven/web-client-ui)
- In the web-client-ui, update `packages/code-studio/package.json` to use the  proxy with the js plugins on port `9010`:
```
"proxy": "http://localhost:9010"
```
- Update `packages/code-studio/.env.development.local` to use the API on port `9000` and set the js plugins path to match the proxy running on port `9010`:
```
REACT_APP_CORE_API_URL=http://localhost:9000/jsapi
REACT_APP_MODULE_PLUGINS_URL=/deephaven-js-plugins
```
- Install web-client-ui dependencies and start the local UI server: `npm i; npm start`
- Open http://localhost:4000 It should connect to the API on port `9000` and load the plugins on port `9010`
- Run sample commands from the [deephaven-plugin-plotly](https://github.com/deephaven/deephaven-plugin-plotly) readme to test plotly integration.
